src/gf.c

```diff
@@ -1997,6 +1997,9 @@ JL_DLLEXPORT void jl_method_instance_add_backedge(jl_method_instance_t *callee,
             jl_gc_wb(callee, backedges);
         }
         else {
+#ifndef JL_NDEBUG
```

Comment:
This introduces a race condition into the code (because of gc, threads, etc). Probably useful to know this scan is costly on some benchmarks though

Reply:
Can you clarify where the race condition is?

Do you mean that inference may simultaneously try to `store_backedges` for the same caller CI from two different threads?

Reply:
Ah, I see, we might be able to now assume that each CI is unique, while in older versions the MI were not expected to be unique